lib/slack-ruby-bot/mvc/controller/base.rb

```diff
@@ -0,0 +1,111 @@
+module SlackRubyBot
+  module MVC
+    module Controller
+      class Base
+        include ActiveSupport::Callbacks
+
+        class << self
+          attr_reader :abstract
+          alias abstract? abstract
+
+          def controllers
+            Base.instance_variable_get(:@controllers)
```

Comment:
Class instance variables are inherited which is one of their advantages over a classic class variable (@ivar vs @@cvar). 

So, the `@controllers` ivar for `SlackRubyBot::MVC::Controller::Base` is *unique* from a `@controllers` ivar in a subclass of it (e.g. `MyController < SlackRubyBot::MVC::Controller::Base`. If we use public class methods to set/get the ivar then we will have multiple `@controller` ivars storing data (one for each subclass). We need to centralize this storage.

So, that's why I am using `Base.instance_variable_{set|get}` here. It ensures that we are always using the `Base` class ivar and centralizing the storage and tracking of the `@command_class` and `@controllers`.

I hope that explanation makes sense. Ask questions if you need more clarity.